packages/contracts/contracts/SmartEscrow.sol

```diff
@@ -99,29 +92,32 @@ contract SmartEscrow is Context, IArbitrable, ReentrancyGuard {
     resolver = _resolver;
     token = _token;
     amounts = _amounts;
-    total = _total;
+    for (uint256 i = 0; i < amounts.length; i++) {
+      total = total.add(amounts[i]);
+    }
     terminationTime = _terminationTime;
     details = _details;
 
     emit Register();
   }
 
-  function confirmLocker() external payable nonReentrant {
+  function confirmLocker() external nonReentrant {
     // client confirms deposit of total & locks in deal
+    // client must have already deposited tokens to the contract
 
     require(!confirmed, "confirmed");
     require(_msgSender() == client, "!client");
 
-    if (token == wETH && msg.value > 0) {
-      require(msg.value == total, "!ETH");
-      IWETH(wETH).deposit();
-      (bool success, ) = wETH.call{value: msg.value}("");
-      require(success, "!transfer");
-      IWETH(wETH).transfer(address(this), msg.value);
-    } else {
-      IERC20(token).safeTransferFrom(msg.sender, address(this), total);
+    if (token == wETH) {
+      uint256 balance = address(this).balance;
+      if (balance > 0) {
+        IWETH(wETH).deposit{value: balance}();
+      }
     }
 
+    uint256 balance = IERC20(token).balanceOf(address(this));
+    require(balance >= total, "not enough token balance");
```

Comment:
the factory version allows client to deposit tokens in multiple transactions and confirm upon deposit